Sources/PostgresNIO/New/PostgresChannelHandler.swift

```diff
@@ -212,23 +212,27 @@ final class PostgresChannelHandler: ChannelDuplexHandler {
                 psqlTask = self.makeStartListeningQuery(channel: channel, context: context)
```

Comment:
we leak the promise here.

Reply:
Not really.
2 things can happen in this function:
1- There are some `return`s. Before all those returns i've added some code to do something with the promise.
2- Otherwise we reach the end of the function where we pass the promise to the `run` function to take care of it.